packages/nitro/src/server/dev.ts

```diff
@@ -128,7 +128,7 @@ export function createDevServer (nitroContext: NitroContext) {
   let watcher: FSWatcher
   function watch () {
     if (watcher) { return }
-    const dReload = debounce(debounce.promise(() => reload().catch(console.warn)), 200, { before: true })
+    const dReload = debounce(() => reload().catch(console.warn), 200, { leading: true })
```

Comment:
As discussed, enabling leading for server has a negative effect for server

```suggestion
    const dReload = debounce(() => reload().catch(console.warn), 100)
```